app/models/manageiq/providers/amazon/inventory_collection_default/cloud_manager.rb

```diff
@@ -176,6 +176,30 @@ def vm_and_template_labels(extra_attributes = {})
       attributes.merge!(extra_attributes)
     end
 
+    def vm_and_template_taggings(extra_attributes = {})
+      # TODO: make a generic Taggings IC and move it to base class?
+      attributes = {
+        :model_class                  => Tagging,
+        :association                  => :vm_and_template_taggings,
+        :manager_ref                  => [:taggable, :tag],
+        :parent_inventory_collections => [:vms, :miq_templates],
+        :inventory_object_attributes  => [
+          :taggable,
+          :tag,
+        ]
+      }
+
+      attributes[:targeted_arel] = lambda do |inventory_collection|
+        ids = inventory_collection.parent_inventory_collections.collect_concat(&:id)
```

Comment:
ids are not available here yet, so we need to use
```manager_uuids = inventory_collection.parent_inventory_collections.collect(&:manager_uuids).map(&:to_a).flatten```

manager uuids are :ems_ref columns of Vm in this case. So we need a nested query that transforms that to ids